compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -67,6 +67,49 @@ getStaticOrReifiedInputDims(OpBuilder &builder, Location loc, Value input,
   return success();
 }
 
+/// gets the rank of the opOperand's type
+static int64_t getRank(OpOperand &opOperand) {
+  ShapedType type = dyn_cast<ShapedType>(opOperand.get().getType());
+  if (!type) {
+    assert(opOperand.get().getType().isIntOrIndexOrFloat() &&
+           "unshaped type should be int or index or float");
+    return 0;
+  }
+  return type.getRank();
+}
+
+/// Method similar to `LinalgOp`s that concatenates shapes of all operands.
+static SmallVector<OpFoldResult>
+createFlatListOfOperandDims(OpBuilder &b, Location loc, Operation *op) {
+  SmallVector<OpFoldResult> res;
+  for (OpOperand &opOperand : op->getOpOperands()) {
+    for (auto dim : llvm::seq(getRank(opOperand))) {
+      res.push_back(linalg::createFoldedDimOp(b, loc, opOperand.get(), dim));
+    }
+  }
+  return res;
+}
+
+/// permutes the offset and size arrays by the result indexes of the provided
+/// affine map
+static SmallVector<Range> getPermutedRange(AffineMap permutation,
+                                           ArrayRef<OpFoldResult> offsets,
+                                           ArrayRef<OpFoldResult> sizes) {
+  auto one = IntegerAttr::get(IndexType::get(permutation.getContext()), 1);
+  assert(permutation.isProjectedPermutation() &&
+         "Indexing map should be a projected permutation");
```

Comment:
This should be `affine map`, but not `indexing map`. Indexing map is the concept of the interface, but not the context to this local function.